src/main/java/edu/kis/vh/nursery/FifoRhymer.java

```diff
@@ -1,18 +1,32 @@
 package edu.kis.vh.nursery;
 
+import edu.kis.vh.nursery.dataStructures.IntArrayStack;
+import edu.kis.vh.nursery.dataStructures.IntLinkedList;
+
 public class FifoRhymer extends DefaultCountingOutRhymer {
 
-	private final DefaultCountingOutRhymer temp = new DefaultCountingOutRhymer();
-	
+	private final IntArrayStack temp = new IntArrayStack();
```

Comment:
No, temp pushes and later pops all values. If Rhymer has more values than temps capacity the operation will fail. Implementation without limitations is needed (IntLinkedlist)